### DISCUSSION TITLE
Adding tests instead of testing against the demo site

### DISCUSSION THREAD

**Discussion by gingerchew** [biati-digital/glightbox#364] [Ideas]
I was messing around with sal.js and needed to add a custom feature.

Working with the tests in the repo made it a lot easier to add my feature and guarantee that the original functionality wasn't getting "stepped on".

[here's the tests](https://github.com/mciastek/sal/tree/master/test)

It is a big undertaking, but I think it would be a big help for future versions and for new contributors/additions.

It would also help automate the testing instead of manually checking against the demo site.

**Comment by biati-digital** [comment]
This is definitely on the list, i tried cypress a few months ago but it was quite slow, i heard really good comments about playwright so i was thinking on trying it.

**Reply by gingerchew** [reply]
I did a group of tests for the functionality that isn't browser related in vitest, and it was blazingly fast. I'll push my local repo so you can take a look at it.

They have a [playwright example](https://github.com/vitest-dev/vitest/tree/main/examples/playwright), but I haven't given it a shot yet.

**Reply by gingerchew** [reply]
Here's the repo I was talking about in... _**OCTOBER?!**_ Boy does time fly: [link](https://github.com/gingerchew/glightbox-testing-example/)

**Reply by biati-digital** [reply]
Thank you, vitest looks great, i think this will be the way to go, i'm currently making some tests and checking your repo.
